railties/lib/rails/application/bootstrap.rb

```diff
@@ -104,6 +105,17 @@ module Bootstrap
       initializer :set_secrets_root, group: :all do
         Rails::Secrets.root = root
       end
+
+      # Sets the Rails.application.url config
+      initializer :initialize_url, group: :all, after: :one do |app|
+        if app.config.application_url
+          url = ActionDispatch::Http::URI.new(app.config.application_url)
+          Rails.application.url = url
+          default_url_options = { host: url.host, protocol: url.scheme, port: url.port }
+          app.default_url_options = default_url_options if app.default_url_options.blank?
+          app.config.action_mailer.default_url_options = default_url_options if app.config.action_mailer.default_url_options.blank?
```

Comment:
```suggestion
          app.config.action_mailer.default_url_options ||= default_url_options
```

I'm not 💯 this actually works but I wonder if it does make this line shorter?